autoload/nerdcommenter.vim

```diff
@@ -1419,46 +1419,19 @@ function! s:PlaceDelimitersAndInsBetween() abort
     let left = s:Left({'space': 1})
     let right = s:Right({'space': 1})
 
-    let theLine = getline('.')
-    let lineHasLeadTabs = s:HasLeadingTabs(theLine) || (theLine =~# '^ *$' && !&expandtab)
-
-    "convert tabs to spaces and adjust the cursors column to take this into
-    "account
-    let untabbedCol = s:UntabbedCol(theLine, col('.'))
-    call setline(line('.'), s:ConvertLeadingTabsToSpaces(theLine))
-    call cursor(line('.'), untabbedCol)
-
-    " get the length of the right delimiter
-    let lenRight = strlen(right)
-
-    let isDelimOnEOL = col('.') >= strlen(getline('.'))
-
-    " if the cursor is in the first col then we gotta insert rather than
-    " append the comment delimiters here
-    let insOrApp = (col('.')==1 ? 'i' : 'a')
-
-    " place the delimiters down. We do it differently depending on whether
-    " there is a left AND right delimiter
-    if lenRight > 0
-        execute ':normal! ' . insOrApp . left . right
-        execute ':normal! ' . lenRight . 'h'
-    else
-        execute ':normal! ' . insOrApp . left
-    endif
-
-    "if needed convert spaces back to tabs and adjust the cursors col
-    "accordingly
-    if lineHasLeadTabs
-        let tabbedCol = s:TabbedCol(getline('.'), col('.'))
-        call setline(line('.'), s:ConvertLeadingSpacesToTabs(getline('.')))
-        call cursor(line('.'), tabbedCol)
-    endif
-
-    if isDelimOnEOL && lenRight ==# 0
-        startinsert!
-    else
-        call feedkeys('a', 'ni')
-    endif
+    " 0. Entered insert normal mode using <C-\><C-O> to maintain the cursor
+    "    position (from <Plug>NERDCommenterInsert).
+    " 1. Enter insert mode without changing the cursor position.
+    "    If the cursor is on EOL (right of the last char), use 'a'.
+    "    Otherwise, use 'i'.
+    let insert = col('.') > strlen(getline('.')) ? 'a' : 'i'
+    " 2. Insert comment delimiters.
+    " 3. Move the cursor to the left of the closing delimiter.
+    " 4. Enter insert normal mode again without changing the cursor position.
+    "    This ensures that returning to the insert mode after finishing the
+    "    script execution does not move the cursor.
+    "             ( 1  )   (    2     )   (                3               )   (      4     )
+    exe 'normal!' insert . left . right . repeat("\<Left>", strchars(right)) . "\<C-\>\<C-O>"
```

Comment:
```suggestion
    execute 'normal!' insert . left . right . repeat("\<Left>", strchars(right)) . "\<C-\>\<C-O>"
```

Just for consistency when searching the code base I'd rather always spell out full command names.